### PR TITLE
Documents

### PR DESCRIPTION
**Pull request by yangarbiter**

### CODE REVIEW

**Comment by yangarbiter** [comment]
@iamyuanchung @tungen 
Please help check with the descriptions in http://libact.readthedocs.org/en/documents/examples/examples.html to see if there is anything need to change.

File: docs/dev_with_libact.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+.. code-block::
+   :lanbuage: python
+
+   super(YourClassName, self).__init__(*args, **kwargs)
+
+This would associate the given dataset with your query strategy and registers
+the update method under the associated dataset as a callback function.
+
+The :py:func:`update` method should be used if the active learning algorithm
+wants to change its internal state after the dataset is updated with newly
```
Comment (tungen) [inline docs/dev_with_libact.rst]: Internal state seems to come out of no where; not mentioned previously

**Comment by tungen** [comment]
Let's start from the Overview...
- libact designed a framework ...  -> libact is a framework...
- libact is constituded by the following parts: -> libact is constituted of the following parts:
- In the start of the problem, ... : not a common used phrase;
- ..., there is a small labeled set, a larger unlabeled set, and a training model: "small" and "larger" may be redundant; oracle is not included
- During the problem, ...: not a common used phrase
- ..., active learning algorithm has to choose a data point... : "has to" is redundant
- In the Labeler/Dataset/QueryStrategy/Model, starting with libact.base.*.* may not be good since libact.base.*.* are not introduced yet. 
- Dataset should be the first submodule to be described IMO.
- As an overview page, the introduction for each submodule seems to be too detailed. To me, a higher level description would be better.
- In the example usage, line 2, qs = QueryStrategy(trn_ds, method='lc') -> the "method = 'lc'" is not needed

**Comment by tungen** [comment]
Next, the Example...
- plot.py is not a good title name; maybe Comparing Different Query Strategies is more appropriate
- In the plot.py subsection, in the example figure, qs Eout renamed as LogisticRegression Eout would be better considering the other legend is "random Eout"
- For both plot.py and label_digits.py, in the full source code, the comments on the top is not needed since those should be pulled out and elaborated in the articles
- label_digits.py is not a good title; Interactive Digits Labeling might be better

**Comment by tungen** [comment]
Next, the Active Learning by Learning section...
- a full page describing the ALBL seems too much and too detailed for me; if you look at scikit-learn User Guide, each module usually only contains a brief/high level description of what it does, the objective function being optimized, its advantage/disadvantage, when should it be/not be used. To me, contents in paragraph 2 and 3 can be condensed.
- variables in IW-ACC like i_t should be defined
- lack example usage

**Comment by tungen** [comment]
Lack something like User Guide in scikit-learn.
Also, we should have a time together to proof read the articles and correct the spelling/grammar errors.

**Comment by tungen** [comment]
[Interactive Digits Labeling ]
- This example simiulates the use case where you want to humane to assign label to active learning algorithm selected samples. We uses the digits dataset provided by sklearn, each time a sample is selected by an active learning algorithm, the sample's image (written digits) will be shown on the screen. The user will have to enter the corresponding digit to the command line, this simulates the labeling process in real world. >> This example simulates the use case where you want users to assign labels to samples selected by active learning algorithms. This example uses the digits dataset provided by sklearn. Each time a sample is selected by an active learning algorithm, the sample (here a written digit) will be shown on the screen. The user will have to enter the corresponding digit to the command line.
- On top, it is the learning curve of ucertainty sampling and random sampling. >> The figure on the top is the learning curve for uncertainty sampling and random sampling.

[ALBL ]
- In ALBL, it utilizes the Exp4.P contextual bandit algorithm. The bandit machines corresponds to each sub-active learning algorithm. It wants to balance between each exploring a good active learning algorithm for this problem and exploiting the active learning algorithms that already performs well to earn a good reward. The reward function that ALBL adopt is the Importance-Weighted-Accuracy (IW-ACC) >> In ALBL, it utilizes the Exp4.P contextual bandit algorithm. Each bandit machine corresponds to one active learning algorithm. ALBL tries to balance between exploring a good active learning algorithm and exploiting the active learning algorithms that already perform well to earn a good reward. The reward function that ALBL adopt is the Importance-Weighted-Accuracy (IW-ACC)
- IW-ACC, 1/nT should be 1/n\tau
- should we have line number for the ALBL example

[API Reference]
- In API references, libact.query_strategies.uncertainty_sampling package:
- This module contains a class that implements two of the most well-known uncertainty sampling query strategies, which are least confidence and smallest margin (margin sampling). >> This module contains a class that implements two of the most well-known uncertainty sampling query strategies: the least confidence method and the smallest margin method (margin sampling).
- In Examples:
    - Here is an example of declaring a HintSVM query_strategy object:
    - qs = UmcertaintySampling(...)

- In API references, libact.query_strategies_variance_reduction package:
- in Parameters/sigma:
    - The regularization term to be added to the diagonal of Fisher information matrix. 1/sigma will be added to the matrix.
- in Parameters/optimality
    - Choosing what to optimize. These options optimize the trace, determinant, and maximum eigenvalue of the inverse Fisher information matrix. -> Optimization target. These options optimize the trace, the determinant, and the maximum eigenvalue of the inverse Fisher information matrix respectively.
- should we add description for class method E, Phi?

- In API references, libact.query_strategies.query_by_committee package:
- Reference is not labeled
- the class methods: disagreement() and teach_students() should add descriptions

- In API references, libact.query_strategies.quire
    - should we add a declaration example for quire?

- In API references, libact.query_strategies.random_sampling
   - should we add a declaration example for random sampling?

**Comment by yangarbiter** [comment]
Variance reduction class method E, Phi are just internal method that shouldn't be used by users.
